postgres/pgdump.go

```diff
@@ -185,13 +185,48 @@ func processStatements(conv *internal.Conv, statements []nodes.Node) *copyOrInse
 			if conv.SchemaMode() {
 				processVariableSetStmt(conv, n)
 			}
+		case nodes.IndexStmt:
+			if conv.SchemaMode() {
+				processIndexStmt(conv, n)
+			}
 		default:
 			conv.SkipStatement(prNodes([]nodes.Node{node}))
 		}
 	}
 	return nil
 }
 
+func toSchemaIndexKeys(indexParams nodes.List) (keys []schema.Key) {
+	var schemaKeys []schema.Key
+	for _, ip := range indexParams.Items {
+		schemaKeys = append(schemaKeys, schema.Key{Column: *ip.(nodes.IndexElem).Name})
+	}
+	return schemaKeys
+}
+
+func processIndexStmt(conv *internal.Conv, n nodes.IndexStmt) {
+	if n.Relation == nil {
+		logStmtError(conv, n, fmt.Errorf("relation is nil"))
+		return
+	}
+	tableName, err := getTableName(conv, *n.Relation)
+	if err != nil {
+		logStmtError(conv, n, fmt.Errorf("can't get table name: %w", err))
+		return
+	}
+	if _, ok := conv.SrcSchema[tableName]; ok {
+		ctable := conv.SrcSchema[tableName]
+		var cols []string
+		for _, ip := range n.IndexParams.Items {
+			cols = append(cols, *ip.(nodes.IndexElem).Name)
+		}
+		ctable.Indexes = append(ctable.Indexes, schema.Index{Name: *n.Idxname, Unique: *&n.Unique, Keys: toIndexKeys(conv, tableName, n.IndexParams.Items)})
```

Comment:
it's easier to read when each field in a struct is defined on a new line -

e.g., 

```
ctable.Indexes = append(ctable.Indexes, schema.Index{
    Name: *n.Idxname,
    Unique: *&n.Unique,
    Keys: toIndexKeys(conv, tableName, n.IndexParams.Items)
})
```

Reply:
updated code as mentioned